ta4j-core/src/main/java/org/ta4j/core/num/NaN.java

```diff
@@ -50,6 +50,16 @@
 
     private NaN() {
     }
+    
+    /**
+     * Returns a {@code Num} version of the given {@code Num}.
+     *
+     * @param val the number
+     * @return the {@code Num}
+     */
+    public static NaN valueOf(NaN val) {
```

Comment:
I would change signature to `valueOf(Number val) -> NaN`

Reply:
Do you mean this:

```
    public static Num valueOf(Number val) {
        return NaN;
    }
```

or that:

```
 public static Num valueOf(Num val) {
        return NaN;
    }
```
?

Reply:
Either we can `valueOf(Num num)` or `valueOf(NaN val)` to be usable for the builder

`BaseBar.builder(org.ta4j.core.num.NaN::valueOf, NaN.class).openPrice(NaN).build();`

I changed the code to use `valueOf(Num num)`.

Reply:
I mean 
```
    public static Num valueOf(Number val) {
        return NaN;
    }
```